### PR TITLE
Task: IOTS-292, Migrating to synapse gateway

### PR DESCRIPTION
**Pull request by amalhub**

### CODE REVIEW

File: components/apimgt-extensions/org.wso2.carbon.apimgt.handlers/src/main/java/org.wso2.carbon.apimgt.handlers/AuthenticationHandler.java [hunk 1]
```diff
@@ -0,0 +1,284 @@
...
+    private static HandlerDescription EMPTY_HANDLER_METADATA = new HandlerDescription("API Security Handler");
+    private HandlerDescription handlerDesc;
+    private ArrayList<String> apiList;
+    private RESTInvoker restInvoker;
+
+    /**
+     * Setting up configurations at the constructor
+     */
+    public AuthenticationHandler() {
+        log.info("Engaging API Security Handler");
```
Comment (geethkokila) [inline components/apimgt-extensions/org.wso2.carbon.apimgt.handlers/src/main/java/org.wso2.carbon.apimgt.handlers/AuthenticationHandler.java:62]: As it seems, this info log will be always printed when the constructed is called. Put this as a debug log

File: components/apimgt-extensions/org.wso2.carbon.apimgt.handlers/src/main/java/org.wso2.carbon.apimgt.handlers/utils/CoreUtils.java [hunk 2]
```diff
@@ -0,0 +1,156 @@
...
+
+public class CoreUtils {
+    private static final Log log = LogFactory.getLog(CoreUtils.class);
+    private static String host = "localhost";
+    private static int httpsPort = 9443;
+    private static String username = "admin";
+    private static String password = "admin";
+    private static String iosVerifyEndpoint = "/api/certificate-mgt/v1.0/admin/certificates/verify/ios";
+    private static String androidVerifyEndpoint = "/api/certificate-mgt/v1.0/admin/certificates/verify/android";
+
```
Comment (geethkokila) [inline components/apimgt-extensions/org.wso2.carbon.apimgt.handlers/src/main/java/org.wso2.carbon.apimgt.handlers/utils/CoreUtils.java:41]: Why do put device type specific things here ?

File: components/apimgt-extensions/org.wso2.carbon.apimgt.handlers/src/main/java/org.wso2.carbon.apimgt.handlers/AuthenticationHandler.java (hunk 1 above)
Comment (amalhub) [reply components/apimgt-extensions/org.wso2.carbon.apimgt.handlers/src/main/java/org.wso2.carbon.apimgt.handlers/AuthenticationHandler.java:62]: As discussed offline, the log only prints at server startup and doesn't trigger upon every request.
It is added to notify developers through the log inspection for any future support analysis. Therefore added a comment in commit: 33422cb

File: components/apimgt-extensions/org.wso2.carbon.apimgt.handlers/src/main/java/org.wso2.carbon.apimgt.handlers/utils/CoreUtils.java (hunk 2 above)
Comment (amalhub) [reply components/apimgt-extensions/org.wso2.carbon.apimgt.handlers/src/main/java/org.wso2.carbon.apimgt.handlers/utils/CoreUtils.java:41]: Refactored the cert verification service to accept a URL param to identify the device type in commit 33422cb.

Thanks

**Pull request closed without merging**